### PR TITLE
Activity class

### PR DESCRIPTION
**Pull request by jonathan-tschida**
#### What's this PR do?
Adds testing and functionality for the Activity class to work with activity data
#### Where should the reviewer start?
Read through the tests and the Activity class and make sure they make sense
#### How should this be manually tested?
Run the test suite 
#### Questions:
- The last three methods are very repetitive, should we refactor them to be one function where you pass in a property key as a parameter?
- Any ideas for another metric we would want to track?

### CODE REVIEW

File: src/Activity.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  getWeek(id, date) {
+    let endDate = this.getUserData(id).findIndex(obj => obj.date === date);
+    return endDate - 6 >= 0 ?
+      this.getUserData(id).slice(endDate - 6, endDate + 1) :
+      this.getUserData(id).slice(0, endDate + 1);
+  }
+
+  getMilesByDay(id, date, userRepo) {
+    let feetWalked = this.getDay(id, date).numSteps * userRepo.getUserData(id).strideLength;
+    return +(feetWalked / 5280).toFixed(1);
```
Comment (dreardon1021) [inline src/Activity.js:23]: Does the plus here just make sure its a number since .toFixed turns it into a string?

File: src/Activity.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      acc + this.getDay(ID, date).numSteps, 0);
+    return Math.round(totalSteps / userIDs.length);
+  }
+
+  getAverageMinutesByDay(date) {
+    let userIDs = [...new Set(this.activityData.map(object => object.userID))];
+    let totalMinutes = userIDs.reduce((acc, ID) =>
+      acc + this.getDay(ID, date).minutesActive, 0);
+    return Math.round(totalMinutes / userIDs.length);
+  }
```
Comment (dreardon1021) [inline src/Activity.js:74]: For our metric we could make it easy and just do total feet walked,
we could do miles walked in a week per day(so an array of numbers for a week [1.2, 4.3, etc],
we could also turn something that involves minutes into hours.

**Review by dreardon1021 — APPROVED** [review]

File: src/Activity.js (hunk 1 above)
Comment (jonathan-tschida) [reply src/Activity.js:23]: Exactly, it's a form of type coercion

File: src/Activity.js (hunk 2 above)
Comment (jonathan-tschida) [reply src/Activity.js:74]: I like the miles per day for a week, I'll see what I can whip up

**Pull request merged**